### DISCUSSION TITLE
left join - code 47 error

### DISCUSSION THREAD

**Discussion by AJPF1** [ClickHouse/ClickHouse#57425] [General]
Hello all,

I have a very simple piece of code which I cannot get to work. I know it's probably something really simple but I just cannot see it: 

The code is: 

**_select timestamp, 
       hash, 
       parseDateTime32BestEffort(JSON_VALUE(json, '$.InitiatedTime')) as InitiatedDateTime, 
       JSON_VALUE(json, '$.ScanType') as ScanType, 
       JSON_VALUE(json, '$.State') as State, 
       JSON_VALUE(json, '$.StateChanged') as StateChanged, 
       JSON_VALUE(json, '$.ThreatLevel') as ThreatLevel,  
       JSON_VALUE(json, '$.WebsiteId') as WebsiteId, 
       last_scan
from [TABLE NAME] as PRIMARY
left join (select JSON_VALUE(json, '$.WebsiteId') as WebsiteId,  
                  max(parseDateTime32BestEffort(JSON_VALUE(json, '$.InitiatedTime')) as InitiatedTime) as last_scan 
           from [TABLE NAME]
           group by WebsiteId) as SECONDARY 
on PRIMARY.WebsiteId = SECONDARY.WebsiteId_**

and the error message is: 

**_SQL Error [47] [07000]: Code: 47. DB::Exception: There's no column 'PRIMARY.WebsiteId' in table 'PRIMARY': While processing PRIMARY.WebsiteId: While processing  LEFT JOIN ...  ON PRIMARY.WebsiteId = SECONDARY.WebsiteId. (UNKNOWN_IDENTIFIER) (version 22.2.2.1)
, server ClickHouseNode_**

I can run the SECONDARY select and it works.  

I can run the first few lines of the code (up to the first **_from_**) and it works. 

For some reason, it just won't do the join for me.

**Comment by den-crane** [comment]
You can use `using` https://fiddle.clickhouse.com/5de47c6a-8850-44f2-911b-648eadb379dd

This behaviour is ansi SQL behavior because there is no WebsiteId column in the table PRIMARY and FROM JOIN ON section works before SELECT.
The same issue will with postgresql.

You should use a subquery for the left table as well.

22.2.2.1 is out of support you should upgrade.
